rfc/0018-certificate-handling.md

```diff
@@ -82,174 +86,199 @@ webhooks:
 
-- Generate Certificate Authority
-- Generate certificate for `kubewarden-controller` and sign it with our CA
-- Start the Kubewarden controller HTTPS server using this certificate
-- Register the webhook against the Kubernetes API, using the certificate of our CA inside of
-  `clientConfig.caBundle`
+- Helm chart generates Certificate Authority(CA) and store it in a secret
+- Helm chart generates cerficate for `kubewarden-controller`, sign it with the
+  CA, and store it in a secret
+- Helm chart uses the CA in the `caBundle` configuration in webhooks to
+  validate CRDs pointing to the `kubewarden-controller`
+- Start the Kubewarden controller HTTPS server using this certificate. The
+  certificate is mounted inside of the Pod
 
 ### New Policy Server Defined
 
 A new `PolicyServer` is defined, the following actions have to be done:
 
 - Generate new certificate, sign it with our CA
-- Create Policy Server Deployment, ensure the certificate is mounted inside of the Pods
+- Create Policy Server Deployment, ensure the certificate is mounted inside of
+  the Pods
 
-This is already done by our current code (Kubewarden 1.6.0).
+It's important to note that the CA used to sign the Policy Server certificate
+is the same CA used to sign the `kubewarden-controller` certificate. It's not
+the internal CA generated to signed Policy Server certificates as it was done
+before.
 
-### New `AdmissionPolicy`/`ClusterAdmissionPolicy` is defined
+### New
+`AdmissionPolicy`/`ClusterAdmissionPolicy`/`ClusterAdmissionPolicyGroup`/`AdmissionPolicyGroup`
+is defined
 
-A new policy is defined inside of the cluster, ensure the following action are done:
+A new policy is defined inside of the cluster, ensure the following action are
+done:
 
-- Register the webhook against the Kubernetes API, use our CA inside of `clientConfig.caBundle`
-
-> **Note:** it doesn't matter which Policy Server is going to host the policy. All the Policy Server
-> instances use a certificate that is signed by our CA. Hence we always put our CA
-> inside of the `clientConfig.caBundle`.
+- Register the webhook against the Kubernetes API, use our CA inside of
+  `clientConfig.caBundle`
 
-This is already done by our current code (Kubewarden 1.6.0).
+> **Note:** it doesn't matter which Policy Server is going to host the policy.
+> All the Policy Server instances use a certificate that is signed by our CA.
+> Hence we always put our CA inside of the `clientConfig.caBundle`.
 
-### Policy Server Certificate Renewed
+### Policy Server and `kubewarden-controller` Certificates Renewed
 
-From time to time we need to renew the certificate issued to a Policy Server instance. The main reason
-to perform this operation is to ensure the Policy Server is not using an expired certificate.
+From time to time we need to renew the certificate issued to the
+`kubewarden-controller` and a Policy Server instance. The main reason to
+perform this operation is to ensure the certicates in use are not expired.
```

Comment:
~when do we do this? we can mention the lookahead time~

nevermind I see it was added to the section below